store/table/index_key_codec.go

```diff
@@ -0,0 +1,66 @@
+package table
+
+// Max255DynamicLengthIndexKeyCodec works with up to 255 byte dynamic size RowIDs.
+// They are encoded as `concat(searchableKey, rowID, len(rowID)[0])`.
```

Comment:
Since coming up with this design I've questioned if length suffixing is really a good idea. It allows for some level of binary prefix scans but is also rather complex and may be error prone. In the SDK it seems like we've settled on length prefixing for bytes and null-termination for strings.